### DISCUSSION TITLE
How do you use custom CSS?

### DISCUSSION THREAD

**Discussion by TheShovel** [TurboWarp/packager#402] [Q&A] [answered]
I've been trying to mess around with custom css with the packager. Sadly, I can't figure out how it works. Can anyone show me some examples of how to use it? Maybe change the transparency of the variable boxes using it or something like that.

**Comment by GarboMuffin** [comment]
The HTML for the monitors looks like

![image](https://user-images.githubusercontent.com/33787854/170730396-57e75959-4d14-4bda-aa35-9f0b602711b8.png)

You can use your browser's developer tools to find this and see which elements make which parts of the interface.

For custom CSS the important part is `class=`. If you want to change the style of `class="sc-monitor-root"`, you could do

```css
.sc-monitor-root {
  background-color: rgba(0, 0, 0, 0.5);
  color: white;
}
```
for example

There are good places online to learn how to use CSS

**Reply by TheShovel** [reply]
thanks a lot!

**Comment by TheShovel** [comment, marked as answer]
I tried doing
```
.sc-monitor-root {
    position: absolute;
    top: 0;
    left: 0;
    /* background: hsla(215, 100%, 95%, 1); */
    color: #575e75;
    /* border: 1px solid hsla(0, 0%, 0%, 0.15); */
    border-radius: 4px;
    font-size: 12px;
    overflow: hidden;
    user-select: none;
    pointer-events: all;
}

.sc-monitor-root[opcode="data_"] .sc-monitor-value-color {
    background-color: #ff8c1a00;
}

.sc-monitor-large-value {
    min-height: 1.4rem;
    min-width: 3rem;
    padding: 0.1rem 0.25rem;
    text-align: center;
    color: black;
    font-size: 1rem;
    white-space: pre-wrap;
}
```
to make the boxes transparent and remove the border, but it doesn't work. It works in chrome with inspect element, but now when exporting with the packager and the custom css.

**Reply by GarboMuffin** [reply]
Custom CSS is added *in addition* to the default styles. If you want to remove something's background use `background: transparent`. You have to override the styles. You might have to add !important in some places.

Custom CSS should only define the things you want to change. Things like the default top: 0 should not be redefined.

**Reply by TheShovel** [reply]
Still doesn't work. The border and the background gets overwrite by the background color option.

**Reply by GarboMuffin** [reply]
then background-color: transparent !important and background: none !important

**Reply by TheShovel** [reply]
still gets overwriten

**Reply by GarboMuffin** [reply]
apparently that can happen, will look later tonight

**Reply by GarboMuffin** [reply]
The packager's default styles now have lower priority than custom CSS

This will work now:

```css
.sc-monitor-value-color {
  background-color: transparent !important;
}
.sc-monitor-root {
  background-color: transparent;
  border: none;
}
```

**Reply by TheShovel** [reply]
Thanks a lot! It worked :D

**Comment by RlabsLLC** [comment]
that doens't seem to work when I package it into a Rasperry Pi app...
